Add quantization and partitioner flow in the qualcomm doc

cccclai opened the pull request:
Summary: Add a session to describe how to lower a model to HTP, including quantization step.

Differential Revision: D78117959

shewu-quic commented on examples/qualcomm/scripts/export_example.py:
```diff
@@ -51,24 +68,38 @@ def main() -> None:
+        use_fp16=use_fp16,
     )
     compile_spec = generate_qnn_executorch_compiler_spec(
         soc_model=QcomChipset.SM8550,
```
Would you like to change here?
```
soc_model=get_soc_to_chipset_map()[args.soc],
```

cccclai replied:
Good catch, updated

shewu-quic reviewed (approved):
Thank you for the effort to make document better!!

metascroy commented on docs/source/backends-qualcomm.md:
````diff
@@ -354,6 +354,94 @@ The model, inputs, and output location are passed to `qnn_executorch_runner` by
+#### Step 2: [Optional] Quantize Your Model
+Choose between quantization approaches, post training quantization (PTQ) or quantization aware training (QAT):
+```python
+from executorch.backends.qualcomm.quantizer.quantizer import QnnQuantizer
````
Is QnnQuantizer configurable?  If so, can we document the configuration?

cccclai replied:
Updated

cccclai committed:
Add quantization and partitioner flow in the qualcomm doc (#12387)

Summary:
Pull Request resolved: https://github.com/pytorch/executorch/pull/12387

Add a session to describe how to lower a model to HTP, including quantization step.

Differential Revision: D78117959
